tests/bluechi_test/bluechilib/bluechictl.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import copy
+import signal
+import subprocess
+import threading
+import time
+import tempfile
+from typing import List, Pattern, Set, Tuple
+
+
+class FileFollower:
+    def __init__(self, file_name):
+        self.pos = 0
+        self.file_name = file_name
+        self.file_desc = None
+
+    def __enter__(self):
+        self.file_desc = open(self.file_name, mode='r')
+        return self
+
+    def __exit__(self, exception_type, exception_value, exception_traceback):
+        if (exception_type):
+            print(f"Exception raised: excpetion_type='{exception_type}', "
+                  f"exception_value='{exception_value}', exception_traceback: {exception_traceback}")
+        if self.file_desc:
+            self.file_desc.close()
+
+    def __iter__(self):
+        while self.new_lines():
+            self.seek()
+            line = self.file_desc.read().split('\n')[0]
+            yield line
+
+            self.pos += len(line) + 1
+
+    def seek(self):
+        self.file_desc.seek(self.pos)
+
+    def new_lines(self):
+        self.seek()
+        return '\n' in self.file_desc.read()
+
+
+class EventEvaluator:
+    def process_line(self, line: str) -> None:
+        pass
+
+    def processing_finished(self) -> bool:
+        pass
+
+
+class SimpleEventEvaluator(EventEvaluator):
+    def __init__(self, expected_patterns: Set[Pattern]):
+        self.expected_patterns = copy.copy(expected_patterns)
+
+    def process_line(self, line: str) -> None:
+        for pattern in copy.copy(self.expected_patterns):
+            if pattern.match(line):
+                print(f"Matched line '{line}'")
+                self.expected_patterns.remove(pattern)
+                break
+
+    def processing_finished(self) -> bool:
+        return len(self.expected_patterns) == 0
+
+
+class BluechiCtlBackgroundRunner:
+    def __init__(self, command: List, evaluator: EventEvaluator):
+        self.command = command
```

Comment:
```suggestion
    def __init__(self, bluechictl_args: List, evaluator: EventEvaluator):
        self.bluechictl = BluechiCtl()
        self.bluechictl_args = bluechictl_args
```
I'd reuse the `BluechiCtl` class here and fix the usage of this background runner to it - currently it could be used to run anything.

Reply:
The run is different in BluechiCtl and in the background runner.
But I can ensure the background runner runs only bluechictl